test/integration/blob.rb

```diff
@@ -53,68 +53,137 @@
+######################                          Create a large block blob                         ######################
+########################################################################################################################
+
+large_blob_name = 'large_test_file.dat'
+begin
+  large_blob_file_name = '/tmp/large_test_file.dat'
+  File.open(large_blob_file_name, 'wb') do |large_file|
+    33.times do
+      large_file.puts(content)
+    end
+  end
+
+  File.open(large_blob_file_name) do |file|
+    options = {
+      key: large_blob_name,
+      body: file
+    }
+    container.files.create(options)
+  end
+ensure
+  File.delete(large_blob_file_name) if File.exist?(large_blob_file_name)
+end
 
 ########################################################################################################################
-######################                          Create a small blob                               ######################
+######################                          Create a small page blob                          ######################
 ########################################################################################################################
 
-small_file_name = 'small_test_file.dat'
-small_blob_name = small_file_name
-content = Array.new(1024 * 1024) { [*'0'..'9', *'a'..'z'].sample }.join
-small_file = File.new(small_file_name, 'w')
-small_file.puts(content)
-small_file.close
-
-storage_data.files.get(container_name, small_blob_name).create(file_path: small_file_name)
+small_page_blob_name = 'small_test_file.vhd'
+content = Array.new(1024 * 1024 + 512) { [*'0'..'9', *'a'..'z'].sample }.join
 
-File.delete(small_file_name)
+options = {
+  key: small_page_blob_name,
+  body: content,
+  blob_type: 'PageBlob'
+}
+container.files.create(options)
 
 ########################################################################################################################
-######################                          Create a large blob                               ######################
+######################                          Create a large page blob                          ######################
 ########################################################################################################################
 
-large_file_name = 'large_test_file.dat'
-large_blob_name = large_file_name
-large_file = File.new(large_file_name, 'w')
-33.times do
-  large_file.puts(content)
+begin
+  large_page_blob_file_name = '/tmp/large_test_file.vhd'
+  large_page_blob_name = 'large_test_file.vhd'
+  File.open(large_page_blob_file_name, 'wb') do |large_file|
+    content = Array.new(1024 * 1024) { [*'0'..'9', *'a'..'z'].sample }.join
+    33.times do
+      large_file.puts(content)
+    end
+    content = Array.new(512) { [*'0'..'9', *'a'..'z'].sample }.join
+    large_file.puts(content)
+  end
+
+  File.open(large_page_blob_file_name) do |file|
```

Comment:
Following exception is raising at this line of code.
The page blob size must be aligned to a 512-byte boundary. But the file size is 34603041.